project/src/com/google/daggerquery/executor/QueryExecutor.java

```diff
@@ -43,8 +44,26 @@ public static void main(String[] args) {
       }
 
       Query query = new Query(args[0], Arrays.copyOfRange(args, 1, args.length));
-      List<String> result = query.execute(new SourcesLoader().loadBindingGraph());
-      result.forEach(printStream::println);
+      List<BindingGraph> bindingGraphs = new SourcesLoader().loadBindingGraphs();
+
+      ImmutableList.Builder resultBuilder = new ImmutableList.Builder();
+
+      // We assume that we successfully executed a query only if in at least one graph it was executed without fail.
+      String failureReason = "";
+      for (BindingGraph bindingGraph: bindingGraphs) {
+        try {
+          resultBuilder.addAll(query.execute(bindingGraph));
+        } catch (IllegalArgumentException e) {
+          failureReason = e.getMessage();
```

Comment:
This variable will get overwritten if there are multiple failures. Is that the intention?

Reply:
If you just care any *any* exception, how about just letting the exception propagate, as I can see you already have an outermost catch block on line 67?

Reply:
If we will ignore this exception catch statement in a line 67 will get it, yes, but a query execution will stop after it. There is might be a situation when we have, for example, **three graphs**, in **two** of them **exception** will be thrown and in **one** of them, the query will be executed **correctly** (because there is no intersection between bindings).

I thought that saving only one (the latest) error message is enough, the user can see others after fixing that one. This exception always points to the missing source node.

Reply:
> If we will ignore this exception catch statement in a line 67 will get it, yes, but a query execution will stop after it.

Ah you're right, thanks! :sweat_smile: